### PR TITLE
Fix/master/az locale

### PR DESCRIPTION
**Pull request by sevdimali**
### What does this changes

Azerbaijani Language Support added.

### CODE REVIEW

File: faker/providers/person/az_AZ/__init__.py [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+
+    formats_male = (
+        '{{first_name_male}} {{last_name_male}}',
+        '{{first_name_male}} {{first_name_male}}',
+        '{{first_name_male}} {{last_name_unisex}}',
+    )
+
+    formats = formats_male + formats_female
+
+    first_names_female = ('Adilə',
```
Comment (fcurella) [inline faker/providers/person/az_AZ/__init__.py:30]: do you have a source for these names?

**Review by fcurella — CHANGES_REQUESTED** [review]
Can you add tests for the new providers?

File: faker/providers/person/az_AZ/__init__.py (hunk 1 above)
Comment (sevdimali) [reply faker/providers/person/az_AZ/__init__.py:30]: I gathered them from different sources, and also made manual changes to them. I will add resources.

**Review by sevdimali — COMMENTED** [review]
needs sources in person provider.

**Review by fcurella — CHANGES_REQUESTED** [review]
Can you fix the linting issues listed in https://github.com/joke2k/faker/pull/1501/checks?check_run_id=3757684760 ?

File: faker/providers/person/az_AZ/__init__.py [hunk 2]
```diff
@@ -0,0 +1,739 @@
+from .. import Provider as PersonProvider
+
```
Comment (fcurella) [inline faker/providers/person/az_AZ/__init__.py:2]: Can you add one more line here?

```suggestion


```

File: tests/providers/test_bank.py [hunk 3]
```diff
@@ -23,12 +24,19 @@ def is_valid_iban(iban):
     return check % 97 == 1
 
 
-def is_valid_aba(aba):
```
Comment (fcurella) [inline tests/providers/test_bank.py:26]: This function is used by other tests. We should keep it

**Review by fcurella — CHANGES_REQUESTED** [review]

File: tests/providers/test_bank.py (hunk 3 above)
Comment (sevdimali) [reply tests/providers/test_bank.py:26]: thanks for the information. Updated.

**Review by fcurella — CHANGES_REQUESTED** [review]
Could you add tests for the `lorem` and the `ssn` providers?

**Comment by sevdimali** [comment]
@fcurella can you please check it? I fixed conflicts.

**Comment by fcurella** [comment]
@sevdimali Tests for the `lorem` and the `ssn` providers are still missing

**Comment by sevdimali** [comment]
@fcurella missing tests added.

File: tests/providers/test_company.py [hunk 4]
```diff
@@ -338,6 +355,7 @@ def test_company_vat(self, faker, num_samples):
             company_vat = faker.company_vat()
             assert self.vat_regex.match(company_vat)
 
+
```
Comment (fcurella) [inline tests/providers/test_company.py:358]: Can you remove this extra blank line?
```suggestion
```

File: faker/providers/ssn/az_AZ/__init__.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+class Provider(SsnProvider):
+    # The FIN code consists of 7 characters (letters and numbers of the English alphabet,
+    # except for the letters "I" and "O").
+
+    characters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'J', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'U', 'V',
+                  'W', 'X', 'Y', 'Z']
+    numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+    all_characters = characters + numbers
+
+    def ssn(self):
```
Comment (fcurella) [inline faker/providers/ssn/az_AZ/__init__.py:13]: Can you add the type hint here?

```suggestion
    def ssn(self) -> str:
```

File: faker/providers/ssn/az_AZ/__init__.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        ssn = ''.join([str(ssn_char) for ssn_char in self.random_elements(elements=self.all_characters, length=7)])
```
Comment (fcurella) [inline faker/providers/ssn/az_AZ/__init__.py:14]: is the casting to `str` necessary here?
```suggestion
        ssn = ''.join(self.random_elements(elements=self.all_characters, length=7))
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    provider_codes = ['50', '51', '55', '60', '70', '77', '99']
+
+    start_digits = ['2', '3', '4', '5', '6', '7', '8', '9']
+
+    area_codes = ['012 3', '012 4', '012 5', '018 6', '020 2', '021 2', '021 4', '022 2', '022 3', '022 4', '023 3',
+                  '024 2', '025 2', '026 2', '026 3', '036 5']
+
+    formats = cellphone_formats + telephone_formats
+
+    def start_digit(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:24]:
```suggestion
    def start_digit(self) -> str:
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        return self.random_element(self.start_digits)
+
+    def provider_code(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:27]:
```suggestion
    def provider_code(self) -> str:
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        return self.random_element(self.provider_codes)
+
+    def area_code(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:30]:
```suggestion
    def area_code(self) -> str:
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        return self.random_element(self.area_codes)
+
+    def cellphone_number(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:33]:
```suggestion
    def cellphone_number(self) -> str:
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        pattern = self.random_element(self.cellphone_formats)
+        return self.numerify(self.generator.parse(pattern))
+
+    def landline_number(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:37]:
```suggestion
    def landline_number(self) -> str:
```

File: faker/providers/phone_number/az_AZ/__init__.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        pattern = self.random_element(self.telephone_formats)
+        return self.numerify(self.generator.parse(pattern))
+
+    def phone_number(self):
```
Comment (fcurella) [inline faker/providers/phone_number/az_AZ/__init__.py:41]:
```suggestion
    def phone_number(self) -> str:
```

**Review by fcurella — CHANGES_REQUESTED** [review]

**Comment by fcurella** [comment]
Thank you!

**Comment by sevdimali** [comment]
thank you too :) 👍

**Pull request merged**